### PR TITLE
[Test Framework] Eliminate use of kubectl

### PR DESCRIPTION
**Pull request by nmittler**
Switching to using the k8s client for all Accessor operations. This eliminates any concern of version issues with kubectl and also supports using custom transports.

### CODE REVIEW

**Comment by ericvn** [comment]
I'll want to test this when running against an IKS cluster. In the past, the k8s client had issues with IKS clusters since the kube config has links to certs. Just want to make sure we fix and/or doc any workarounds if needed.

**Comment by nmittler** [comment]
@ericvn you should be able to customize the transport used now that [this change]
(https://github.com/nmittler/istio/commit/8e9621ab76958246565f4ddcb303c23d256d23c7) was submitted.

**Comment by ericvn** [comment]
I did run some tests against an IKS cluster and didn't see the issue I was worried about.

**Comment by mandarjog** [comment]
@ostromart are these changes similar to the change that went into the operator while eliminating kubectl?

File: pkg/test/kube/accessor.go [hunk 1]
```diff
@@ -629,55 +651,477 @@ func (a *Accessor) GetUnstructured(gvr schema.GroupVersionResource, namespace, n
...
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := a.clientFactory.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+
+	// Create the options.
+	streams, _, stdout, stderr := genericclioptions.NewTestIOStreams()
```
Comment (howardjohn) [inline pkg/test/kube/accessor.go:730]: There is a lot of magic boilerplate here. I assume this is copy+pasted from kubectl code or something? Are there any higher level libraries that allow applying arbitrary objects?

I know the operator is doing this, so its possible we can look there for inspiration and/or code reuse

**Pull request merged**